test/integration/common_functions.sh

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+## VARIABLES ------------------------------------------------------- ##
+
+declare -A DEPLOYMENT_RES=(["ADMIT"]="0" ["REJECT"]="0")
+RED="\033[0;31m"
+GREEN="\033[0;32m"
+NC="\033[0m"
+SUCCESS="${GREEN}SUCCESS${NC}"
+FAILED="${RED}FAILED${NC}"
+RETRY=3
+
+## UTILS ------------------------------------------------------- ##
+fail() {
+    echo -e "${FAILED}"
+    exit 1
+}
+
+success() {
+    echo -e "${SUCCESS}"
+}
+
+null_to_empty() {
+    read in
+
+    if [[ "$in" == "null" ]]; then
+        echo ""
+    else
+        echo "$in"
+    fi
+}
+
+ghcr_case() {
+    if [[ -n "${IMAGE+x}" && -n "${IMAGEPULLSECRET+x}" && "${GITHUB_WORKFLOW}" != "release" ]]; then
+        return 0
```

Comment:
I think this is somewhat counter-intuitive with exit codes being the exact opposite of this, but maybe that's just me

Reply:
it is how it is 🤷